crates/solver/src/auction_preprocessing.rs

```diff
@@ -1,345 +1,9 @@
-        .flat_map(|order| [order.sell_token, order.buy_token])
-        .filter(|token| *token != native_token)
-        .collect::<HashSet<_>>()
-        .into_iter()
-        .map(|token| price_estimation::Query {
-            // For ranking purposes it doesn't matter how the external price vector is scaled,
-            // but native_token is used here anyway for better logging/debugging.
-            sell_token: native_token,
-            buy_token: token,
-            in_amount: native_token_amount_to_estimate_prices_with,
-            kind: model::order::OrderKind::Sell,
-        })
-        .collect::<Vec<_>>();
-    let estimates = price_estimator.estimates(&queries).await;
-
-    fn log_err(token: H160, err: &str) {
-        tracing::warn!("failed to estimate price for token {}: {}", token, err);
-    }
-    let mut prices: HashMap<_, _> = queries
-        .into_iter()
-        .zip(estimates)
-        .filter_map(|(query, estimate)| {
-            let estimate = match estimate {
-                Ok(estimate) => estimate,
-                Err(err) => {
-                    log_err(query.buy_token, &format!("{:?}", err));
-                    return None;
-                }
-            };
-            let price = match estimate.price_in_sell_token_rational(&query) {
-                Some(price) => price,
-                None => {
-                    log_err(query.buy_token, "infinite price");
-                    return None;
-                }
-            };
-            Some((query.buy_token, price))
-        })
-        .collect();
-
-    // Always include the native token.
-    prices.insert(native_token, num::one());
-    // And the placeholder for its native counterpart.
-    prices.insert(BUY_ETH_ADDRESS, num::one());
-
-    // Derive exchange rate from orders if only one token pair is traded
-    augment_prices(orders, &mut prices);
-
-    prices
-}
-
-// Filter limit orders for which we don't have price estimates as they cannot be considered for the objective criterion
-pub fn orders_with_price_estimates(
-    orders: Vec<LimitOrder>,
-    prices: &HashMap<H160, BigRational>,
-) -> Vec<LimitOrder> {
-    let (orders, removed_orders): (Vec<_>, Vec<_>) = orders.into_iter().partition(|order| {
-        prices.contains_key(&order.sell_token) && prices.contains_key(&order.buy_token)
-    });
-    if !removed_orders.is_empty() {
-        tracing::debug!(
-            "pruned {} orders: {:?}",
-            removed_orders.len(),
-            removed_orders,
-        );
-    }
-    orders
-}
-
-// In case we cannot estimate the price for a token but there is only one pair traded in the batch the accuracy of the price estimate for ranking a solution doesn't matter that much (surplus is not competing with other tokens).
-// Therefore we can deduct a price estimate by looking at the average limit price of the orders on the single token pair
-fn augment_prices(orders: &[LimitOrder], prices: &mut HashMap<H160, BigRational>) {
```

Comment:
I think this currently doesn't happen in the `/auction` endpoint.
This optimization doesn't seem very important to me so I would be fine with dropping it completely. I just thought that this should be highlighted. Maybe somebody else has a different opinion. 🤷‍♂️

Reply:
Sorry! There is some missing context that I had included in #1649:

> We no longer need to augment prices in the driver (since, not only do we use the native token estimator which supports more tokens, additionally orders for which there are no native prices shouldn't get added to the orderbook anymore, and finally we also filter the orders before sending them to the driver so augmenting would never really do anything anyway).

Reply:
Forgot about that. Thanks for reiterating. 👍